plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/session/Session.kt

```diff
@@ -114,15 +114,40 @@ class Session(val tabID: String, val project: Project) {
     /**
      * Triggered by the Insert code follow-up button to apply code changes.
      */
-    suspend fun insertChanges(
+    suspend fun insertChangesAndUpdateFileComponents(
+        filePaths: List<NewFileZipInfo>,
+        deletedFiles: List<DeletedFileInfo>,
+        references: List<CodeReferenceGenerated>,
+        messenger: MessagePublisher,
+    ) {
+        insertChangesWithoutUpdateFileComponents(filePaths, deletedFiles, references, messenger)
+        val codeResultMessageId = this._codeResultMessageId
+        if (codeResultMessageId != null) {
+            messenger.updateFileComponent(this.tabID, filePaths, deletedFiles, codeResultMessageId)
+        }
+    }
+
+    suspend fun insertChangesWithoutUpdateFileComponents(
```

Comment:
I'm going to add comments to make sure I understand the roles of each component. 

This is used to 
1. write files to disk
2. log any references

Is that right?